DEVELOPMENT.md

```diff
@@ -1,5 +1,14 @@
 # Development
 
+## Proxy-Wasm ABI compliance & trust model
+
+It is expected that the host environment is completely trusted and compliant
+with the [Proxy-Wasm ABI specification] (currently v0.2.1).
+
+This means that receiving unexpected callback, status, lifecycle flow and/or
+incorrectly serialized parameters is considered a permanent failure and will
```

Comment:
optional: "[...] incorrectly serialized parameters" -> "incorrectly serialized parameters in callbacks from the host"
(to make directionality clearer)

Reply:
Done (sans "in callbacks", since values returned from hostcalls could be also corrupted).